File: [탐색] 네트워크.cpp
#include <string>
#include <vector>
using namespace std;

#define INT_VECTOR vector<int>

void reculsiveDFS(int index, vector<INT_VECTOR> map, vector<bool> & visit) {
    
    visit[index] = false;
    for (int ii = 0; ii < map.size(); ii++) { 
        if (map[index][ii] == 1 && visit[ii]) { reculsiveDFS(ii, map, visit); }   
    }
    
}

int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    
    vector<bool> visit = vector<bool>(computers.size(), true);
    
    int length = computers.size();
    for (int ii = 0; ii < length; ii++) {
        if (visit[ii]) { reculsiveDFS(ii, computers, visit); answer++; }
    }
    
    return answer;
}

File: [탐색] 네트워크.swift
import Foundation

let CONNECTED = 1

func findConnected(visited: inout [Bool], map: [[Int]], index: Int, size: Int) {
    
    visited[index] = true
    
    for ii in 0..<size {
        let there = map[index][ii]
        if visited[ii] == false && there == CONNECTED {
            findConnected(visited: &visited, map: map, index: ii, size: size)
        }
    }
}

func solution(_ n:Int, _ computers:[[Int]]) -> Int {
    
    var visited: [Bool] = Array(repeating: false, count: n)
    
    var answer = 0
    for index in 0..<n where visited[index] == false {
        answer = answer + 1
        findConnected(visited: &visited, map: computers, index: index, size: n)
    }
    
    return answer
}
